### PR TITLE
i8: add Min/Max/Clamp/Abs/Neg element-wise primitives

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds the Tier 2 element-wise group from the int8/byte tracking issue (#132): five single-pass primitives that round out the `i8` toolkit, each a single hardware instruction (or a short saturating construction) per lane.

- `Min(dst, a, b)` / `Max(dst, a, b)`: element-wise signed minimum/maximum of two slices (`VPMINSB`/`VPMAXSB`; `SMIN`/`SMAX` on NEON). Distinct from the existing `MinMax` reduction.
- `Clamp(dst, src, lo, hi)`: clamp each element to `[lo, hi]` for activation clipping (broadcast bounds, then max-then-min; `VPBROADCASTB`+`VPMAXSB`+`VPMINSB`, `DUP`+`SMAX`+`SMIN` on NEON). With `lo > hi` every element maps to `hi`, documented and tested.
- `Abs(dst, a)`: saturating absolute value, so `abs(-128) = 127` (`max(a, saturating(0-a))` on AVX2; `SQABS` on NEON). `VPABSB` is deliberately not used because it wraps `-128`.
- `Neg(dst, a)`: saturating negation, so `neg(-128) = 127` (`saturating(0-a)` via `VPSUBSB` on AVX2; `SQNEG` on NEON).

Each ships the established discipline: a pure-Go reference (the source of truth), AVX2 + NEON kernels validated against it, parity tests across a length sweep, literal saturation edge cases, zero-allocation and trailing-capacity assertions, a differential fuzz target, and benchmarks. All NEON instructions are hand-encoded `WORD` directives whose encodings were verified with `aarch64-linux-gnu-as`/`objdump`, cross-checked against `arm64asm` by the repo's `asmcheck` harness, and independently confirmed against the Arm ARM. New AVX2 kernels use only mnemonics the Go assembler emits directly (no hand-encoding).

Zero allocations and bit-exact parity with the reference on both architectures. The remaining Tier 2 items (`AbsDiff`, scalar-broadcast `AddScalarSaturate`/`SubScalarSaturate`) and the Tier 1 ML primitives are left for follow-up PRs.

## Related Issues

Relates to #132 (implements the "Tier 2 element-wise (Min/Max/Clamp/Abs/Neg)" group).

## Test Plan

- [x] `go test ./i8/ -count=1` passes on amd64 (AVX2 path and the `< 32`-byte Go-reference tail).
- [x] `SIMD_DISABLE=avx go test ./i8/` passes (pure-Go reference path validated independently).
- [x] Cross-compiled and run natively on aarch64 (Raspberry Pi 5): parity, zero-alloc, trailing-capacity, and fuzz-seed tests pass on the NEON path.
- [x] `SIMD_REQUIRE_OBJDUMP=1 go test . -run TestArm64WordEncodings` cross-checks every new `WORD` against `arm64asm`/objdump.
- [x] `go vet ./i8/` and `GOARCH=arm64 go vet ./i8/` clean (asm arg offsets).
- [x] `go test ./i8/ -fuzz FuzzI8Elementwise` ran ~5M executions with no failures.
- [x] `golangci-lint run ./i8/...` clean.
- [x] `go test ./... -count=1` (full repo) and `go test ./i8/ -race` pass.
- [x] Benchmarks confirm 0 B/op: AVX2 ~50-70 GB/s, NEON (rpi5) ~16-21 GB/s.

## Summary by CodeRabbit

* **New Features**
  * Added five int8 element-wise operations with SIMD acceleration (AVX2, NEON): `Min`, `Max`, `Clamp`, `Abs`, and `Neg` with saturating behavior for edge cases.

* **Documentation**
  * Updated package documentation and README to describe new int8 operations and their semantics.

* **Tests**
  * Added unit tests, benchmarks, and fuzz tests for all new operations.

### CODE REVIEW

File: docs/superpowers/plans/2026-06-13-i8-elementwise-tier2.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+- [ ] **Step 4: Full amd64 suite** `go test ./i8/ -count=1` and a short fuzz `go test ./i8/ -run x -fuzz FuzzI8Elementwise -fuzztime 20s`. Expected: PASS, no new failures.
+
+- [ ] **Step 5: Commit** `test(i8): parity, fuzz, benchmarks for Min/Max/Clamp/Abs/Neg`.
+
+## Task 5: Cross-arch verification + docs
+
+- [ ] **Step 1: Cross-compile the i8 test binary and run on rpi5** (native aarch64):
+  `GOARCH=arm64 go test -c -o /tmp/i8.arm64.test ./i8 && scp /tmp/i8.arm64.test rpi5.local:/tmp/ && ssh rpi5.local '/tmp/i8.arm64.test -test.run "Min|Max|Clamp|Abs|Neg|ZeroAlloc|Trailing|Sum|Dot|MinMax|ToInt" -test.v'`. Expected: PASS (exercises both the SDOT-capable and base NEON paths).
+- [ ] **Step 2: Benchmark on amd64 and rpi5**, capture `-benchmem` to confirm 0 allocs and record speedups for the PR body.
+- [ ] **Step 3: `golangci-lint run ./i8/...`** clean.
```
[the remark this answers is not in the document]
Comment (tphakala) [reply docs/superpowers/plans/2026-06-13-i8-elementwise-tier2.md]: Removed the plan doc from this PR rather than tweaking the lint scope. docs/superpowers/plans/ holds scratch task lists; the repo tracks committed design under docs/superpowers/specs/, so the plan doesn't belong in the library tree.

File: docs/superpowers/plans/2026-06-13-i8-elementwise-tier2.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+---
+
+## Self-review notes
+
+- Spec coverage: implements exactly the issue #132 "Tier 2 element-wise (Min/Max/Clamp/Abs/Neg)" group. `AbsDiff` and `AddScalarSaturate`/`SubScalarSaturate` are deliberately deferred to a follow-up PR to keep this one cohesive.
+- Naming collision resolved: arch dispatch entry for clamp is `clampElemI8` (not `clampI8`, which is the existing scalar saturate helper). `absInt` added as a small helper; confirm no existing `absInt` in package before adding.
+- Saturation correctness: `Abs`/`Neg` use saturating ops/refs so `-128` maps to `127`; this is the one behavior that must have explicit literal tests and a fuzz check.
+</content>
+</invoke>
```
[the remark this answers is not in the document]
Comment (tphakala) [reply docs/superpowers/plans/2026-06-13-i8-elementwise-tier2.md]: Good catch. Removed the plan doc entirely, which drops the leaked template-tag footer.
